js/main.js

```diff
@@ -1,27 +1,154 @@
-
 /* global data */
 /* exported data */
 var $imageUrl = document.querySelector('.image-url');
 var $userPhotoUrl = document.querySelector('#user-photoUrl');
 var $newEntry = document.querySelector('.journal-entry-form');
+var $newEntryBtn = document.querySelector('.new-button');
+var $navAnchor = document.querySelector('.entries-nav-link');
+var $viewEntryForm = document.querySelector('.view-entry-form');
+var $viewEntries = document.querySelector('.entries-view');
+var $newUlEntries = document.querySelector('.list-entries');
+var $formheader = document.querySelector('.form-header');
+var dataEntryId = null;
+var journalEntryId = null;
+$userPhotoUrl.addEventListener('input', entryImageUpdate);
+$newEntry.addEventListener('submit', saveEntry);
+$newEntryBtn.addEventListener('click', viewEntryForm);
+$navAnchor.addEventListener('click', viewEntries);
+$newUlEntries.addEventListener('click', editEntry);
+window.addEventListener('DOMContentLoaded', entryLoad);
+
+if (data.view === 'entry-form') {
+  $viewEntryForm.className = 'view-entry-form';
+  $viewEntries.className = 'hidden entries-view';
+} else {
+  $viewEntryForm.className = 'hidden view-entry-form';
+  $viewEntries.className = 'entries-view';
+}
 
 function entryImageUpdate(event) {
   var $photoInputValue = event.target.value;
   $imageUrl.setAttribute('src', $photoInputValue);
 }
-
-$userPhotoUrl.addEventListener('input', entryImageUpdate);
-
 function saveEntry(event) {
   event.preventDefault();
-  var userInput = {};
-  userInput.title = $newEntry.elements.title.value;
-  userInput.photo = $newEntry.elements.photo.value;
-  userInput.notes = $newEntry.elements.notes.value;
-  userInput.entryId = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(userInput);
+  if (data.editing === null) {
+    var userInput = {};
+    userInput.title = $newEntry.elements.title.value;
+    userInput.photo = $newEntry.elements.photo.value;
+    userInput.notes = $newEntry.elements.notes.value;
+    data.entries.unshift(userInput);
+    userInput.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    var newNode = createEntry(data.entries[0]);
+    $newUlEntries.append(newNode);
+  } else {
+    data.editing.title = $newEntry.elements.title.value;
+    data.editing.photo = $newEntry.elements.photo.value;
+    data.editing.notes = $newEntry.elements.notes.value;
+    for (var x = 0; x < data.entries.length; x++) {
+      if (data.entries[x].entryId === data.editing.entryId) {
+        data.entries[x] = data.editing;
+      }
+      var editEntryItem = createEntry(data.editing);
+      var $journalNode = document.querySelectorAll('.journal-entry');
+      journalEntryId = $journalNode[x].getAttribute('data-entry-id');
```

Comment:
You wouldn't want to do this for every entry on the page, just the one that matches. You should move this into the conditional above.